src/ClockPanel.tsx

```diff
@@ -10,110 +10,72 @@ import './external/moment-duration-format';
 import { getTemplateSrv } from '@grafana/runtime';
 
 interface Props extends Themeable2, PanelProps<ClockOptions> {}
-interface State {
-  // eslint-disable-next-line
-  now: Moment;
-}
 
 export function getTimeZoneNames(): string[] {
   return (moment as any).tz.names();
 }
 
-class UnthemedClockPanel extends PureComponent<Props, State> {
-  timerID?: any = 0;
-  state = { now: this.getTZ(), timezone: '' };
-
-  componentDidMount() {
-    this.initTimers();
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    const { options, data } = this.props;
-    const { options: prevOptions, data: prevData } = prevProps;
-
-    if (options.refresh !== prevOptions.refresh) {
-      this.initTimers();
-    }
-
-    if (prevData !== data) {
-      this.onPanelRefresh();
-    }
+function getTZ(tz?: string): Moment {
+  if (!tz) {
+    tz = (moment as any).tz.guess();
+  } else {
+    tz = getTemplateSrv().replace(tz);
   }
+  return (moment() as any).tz(tz);
+}
 
-  initTimers = () => {
-    const { refresh } = this.props.options;
-
-    if (this.timerID) {
-      clearInterval(this.timerID);
-      this.timerID = 0;
-    }
-
-    if (refresh === ClockRefresh.dashboard) {
-      return this.tick();
-    }
-
-    const delay = 1000; // 1sec
-    this.timerID = setInterval(
-      () => this.tick(),
-      delay // 1 second or 1 min
-    );
-  };
-
-  onPanelRefresh = () => {
-    const { refresh } = this.props.options;
-    if (refresh === ClockRefresh.dashboard) {
-      this.tick();
-    }
-  };
-
-  componentWillUnmount() {
-    if (this.timerID) {
-      clearInterval(this.timerID);
-      this.timerID = 0;
-    }
+function getTimeFormat(clockType: ClockType, timeSettings: TimeSettings): string {
+  if (clockType === ClockType.Custom && timeSettings.customFormat) {
+    return timeSettings.customFormat;
   }
 
-  tick() {
-    const { timezone } = this.props.options;
-    this.setState({ now: this.getTZ(timezone) });
+  if (clockType === ClockType.H12) {
+    return 'h:mm:ss A';
   }
 
-  getTimeFormat() {
-    const { clockType, timeSettings } = this.props.options;
+  return 'HH:mm:ss';
+}
 
-    if (clockType === ClockType.Custom && timeSettings.customFormat) {
-      return timeSettings.customFormat;
+export function UnthemedClockPanelFunctional(props: Props) {
+  const [now, setNow] = useState<Moment>(getTZ());
+  const { options, width, height, theme } = props;
+  const { timezone, dateSettings, timezoneSettings, bgColor } = options;
+
+  const className = css`
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    flex-flow: column wrap;
+    text-align: center;
+    background-color: ${!bgColor ? theme.colors.background.primary : theme.v1.visualization.getColorByName(bgColor)};
+  `;
+
+  // Clock refresh only on dashboard refresh
+  useEffect(() => {
+    if (props.options.refresh === ClockRefresh.dashboard) {
+      setNow(getTZ(props.options.timezone));
     }
+  }, [props]);
```

Comment:
I tested it and it even it only re-renders if the dashboard changes (e.g. time range) or the user clicks the refresh button.

This is because the props the dashboard passes to the component only changes in those specific cases, not on every render of the panels.

Reply:
Ok ok, is this also true if we have things like template variables, annotations etc. setup in the dashboard? If that is the case then I'm happy with the current solution 👍🏻

Reply:
Yes. If they don't change, it won't re-render.